src/charm.py

```diff
@@ -370,11 +383,50 @@ def _on_relation_database_changed(self, event: MySQLDatabaseChangedEvent) -> Non
             event: An instance of opslib.mysql.MySQLDatabaseChangedEvent represents the new
                 database connection information.
         """
+        logger.warning(
+            "The `db` relation is marked for deprecation. "
+            "Please use `database` relation instead."
+        )
         self.state.relation_db_host = event.host
         self.state.relation_db_name = event.database
         self.state.relation_db_user = event.user
         self.state.relation_db_password = event.password
 
+    def _parse_database_endpoints(self, endpoints: str | None) -> str | None:
+        """Retrieve a single database endpoint from database endpoints.
+
+        Args:
+            endpoints: Comma separated endpoints of format host:port
+
+        Returns:
+            Hostname of database running on port 3306. None if no endpoints are provided.
+            Note that WordPress will throw MySQL Error when supplying a port with the hostname
+            (i.e. host:port).
+        """
+        if not endpoints:
+            return None
+        urls = endpoints.split(",")
+        return urls[0].split(":")[0]
+
+    def _on_database_created(self, event: DatabaseCreatedEvent) -> None:
+        """Store database connection configuration details to charm state.
+
+        Args:
+            event: Event emitted when a new database is created for use on this relation,
+            propagated from relation_changed event under the hood.
+        """
+        self.state.relation_db_host = self._parse_database_endpoints(endpoints=event.endpoints)
+        self.state.relation_db_name = event.database
```

Comment:
I wanted to keep the existing method of reconciliation as is for now and have a major refactor on the next ticket. But it seems that it's necessary to refactor this piece of code now since using the same state will cause bugs when both db relations are related and one departs.
Thank you for the comment :)